### PR TITLE
Generate correct output tensor names in C Interface API

### PR DESCRIPTION
**Pull request by grant-arm**
Currently for the C Interface API, output tensor names default to "output", "output0", "output1" etc.
This is in contrast to the input tensor names that are derived from the model.

This PR updates the code to generate output tensor names derived from the model and updates unit tests accordingly.

### CODE REVIEW

**Comment by grant-arm** [comment]
@manupa-arm @Mousius @leandron @areusch

File: python/tvm/relay/frontend/tflite.py [hunk 1]
```diff
@@ -3769,6 +3770,15 @@ def from_tflite(model, shape_dict=None, dtype_dict=None, op_converter=OperatorCo
     params = {k: _nd.array(np.array(v)) for k, v in exp_tab.params.items()}
     outputs = [exp_tab.get_expr(get_tensor_name(subgraph, i)) for i in model_outputs]
     outputs = outputs[0] if len(outputs) == 1 else _expr.Tuple(outputs)
-    func = _function.Function(analysis.free_vars(outputs), outputs)
+    attrs = tvm.ir.make_node(
+        "DictAttrs",
+        **{
+            "output_tensor_names": [
+                re.sub(r"\W", "_", get_tensor_name(subgraph, model_output))
```
Comment (Mousius) [inline python/tvm/relay/frontend/tflite.py:3777]: Can we use `sanitize_name` from https://github.com/apache/tvm/blob/main/python/tvm/relay/backend/name_transforms.py#L102 here?

**Review by Mousius — CHANGES_REQUESTED** [review]

**Comment by Mousius** [comment]
Overall looks ok to me @grant-arm , would just be good to use the name transformation functions if we can :smile_cat:

File: python/tvm/relay/frontend/tflite.py (hunk 1 above)
Comment (grant-arm) [reply python/tvm/relay/frontend/tflite.py]: Thanks @Mousius , I've changed that now.

**Review by Mousius — APPROVED** [review]

**Comment by Mousius** [comment]
Fantastic improvement @grant-arm!

**Pull request merged**